.github/workflows/precompiled.yaml

```diff
@@ -70,6 +86,14 @@ jobs:
           echo "PUSH_ON_BUILD=${GENERATE_ARTIFACTS}" >> $GITHUB_ENV
           echo "BUILD_MULTI_ARCH_IMAGES=${GENERATE_ARTIFACTS}" >> $GITHUB_ENV
 
+          if [ "$DIST" == "ubuntu22.04" ]; then
+            echo  "BASE_TARGET=jammy" >> $GITHUB_OUTPUT
+            echo "LTS_KERNEL=5.15" >> $GITHUB_OUTPUT
+          elif [ "$DIST" == "ubuntu24.04" ]; then
+            echo  "BASE_TARGET=noble" >> $GITHUB_OUTPUT
+            echo "LTS_KERNEL=6.8" >> $GITHUB_OUTPUT
+          fi
```

Comment:
To avoid duplicating this, what about pushing this logic to the Makefile / Dockerfile? That way we do not need to set these variables when running `make build-base-${BASE_TARGET}`

In the Makefile, we can define the `BASE_TARGET` variable for the respective distro targets:
```
build-signed_ubuntu22.04%: BASE_TARGET = jammy

build-signed_ubuntu24.04%: BASE_TARGET = noble
``` 

For `LTS_KERNEL`, we can set a default value for this ARG in the Dockerfile so that we do not need to set anything here. For example, for the jammy base image, we would have:
```
ARG LTS_KERNEL=5.15
ENV LTS_KERNEL=${LTS_KERNEL}
```

Reply:
moved LTS_KERNEL and BASE_TARGET   to set-driver-version-matrix   ( generic solution)
 
 `  # BASE_TARGET setup
          declare -A MAP_BASE_TARGETS
          MAP_BASE_TARGETS["ubuntu22.04"]="jammy"
          MAP_BASE_TARGETS["ubuntu24.04"]="noble"
          BASE_TARGETS_JSON="{"
          for key in "${!MAP_BASE_TARGETS[@]}"; do
            BASE_TARGETS_JSON+="\"$key\":\"${MAP_BASE_TARGETS[$key]}\","
          done
          BASE_TARGETS_JSON="${BASE_TARGETS_JSON%,}}"
          echo "BASE_TARGETS=$BASE_TARGETS_JSON" >> "$GITHUB_OUTPUT"

          # LTS_KERNELS env setup
          declare -A MAP_LTS_KERNELS
          MAP_LTS_KERNELS["ubuntu22.04"]="5.15"
          MAP_LTS_KERNELS["ubuntu24.04"]="6.8"
          LTS_KERNELS_JSON="{"
          for key in "${!MAP_LTS_KERNELS[@]}"; do
            LTS_KERNELS_JSON+="\"$key\":\"${MAP_LTS_KERNELS[$key]}\","
          done
          LTS_KERNELS_JSON="${LTS_KERNELS_JSON%,}}"
          echo "LTS_KERNELS=$LTS_KERNELS_JSON" >> "$GITHUB_OUTPUT"`
          
          extraction 
          
           ```
 BASE_TARGETS_JSON='${{ needs.set-driver-version-matrix.outputs.BASE_TARGETS }}'
          BASE_TARGET=$(echo "$BASE_TARGETS_JSON" | jq -r --arg dist "$DIST" '.[$dist]')
          echo "BASE_TARGET=$BASE_TARGET" >> $GITHUB_ENV
          LTS_KERNELS_JSON='${{ needs.set-driver-version-matrix.outputs.LTS_KERNELS }}'
          LTS_KERNEL=$(echo "$LTS_KERNELS_JSON" | jq -r --arg dist "$DIST" '.[$dist]')
          echo "LTS_KERNEL=$LTS_KERNEL" >> $GITHUB_ENV
          
          
            BASE_TARGET: ${{ env.BASE_TARGET }}
          LTS_KERNEL: ${{ env.LTS_KERNEL }}
```

moved as we are using it at multiple : make and finding kernel version also . we need to change it all those places . 

make DRIVER_BRANCH=${{ matrix.driver_branch }} KERNEL_FLAVOR=${{ matrix.flavor }} LTS_KERNEL=${LTS_KERNEL} build-base-${BASE_TARGET}

          trap "docker rm -f base-${BASE_TARGET}-${{ matrix.flavor }}" EXIT
          docker run -d --name base-${BASE_TARGET}-${{ matrix.flavor }} ghcr.io/nvidia/driver:base-${BASE_TARGET}-${{ matrix.flavor }}-${{ matrix.driver_branch }}
          
          findkernelversion.sh also using it

Reply:
TODO:- added for next PR

Reply:
@cdesiniotis 

removed map based BASE_TARGET and LTS_KERNEL and followed much cleaner approach  :
definition: 
      base_target_ubuntu22_04: ${{ steps.extract_driver_branch.outputs.base_target_ubuntu22_04 }}
      lts_kernel_ubuntu22_04: ${{ steps.extract_driver_branch.outputs.lts_kernel_ubuntu22_04 }}
      BASE_TARGET_ubuntu24_04: ${{ steps.extract_driver_branch.outputs.base_target_ubuntu24_04 }}
      lts_kernel_ubuntu24_04: ${{ steps.extract_driver_branch.outputs.lts_kernel_ubuntu24_04 }} 

set : 
   # BASE_TARGET and LTS_KERNEL setup
   echo "base_target_ubuntu22_04=jammy" >> $GITHUB_OUTPUT
   echo "base_target_ubuntu24_04=noble" >> $GITHUB_OUTPUT
   echo "lts_kernel_ubuntu22_04=5.15" >> $GITHUB_OUTPUT
   echo "lts_kernel_ubuntu24_04=6.8" >> $GITHUB_OUTPUT

extraction: 
  VAR_NAME_BASE_TARGET=$(echo BASE_TARGET_${DIST} | sed 's/\./_/g')
  echo "VAR_NAME_BASE_TARGET=${VAR_NAME_BASE_TARGET}" >> $GITHUB_ENV
  VAR_NAME_LTS_KERNEL=$(echo LTS_KERNEL_${DIST} | sed 's/\./_/g')
  echo "VAR_NAME_LTS_KERNEL=${VAR_NAME_LTS_KERNEL}" >> $GITHUB_ENV


    env:
          BASE_TARGET: ${{ needs.set-driver-version-matrix.outputs[env.VAR_NAME_BASE_TARGET] }}
          LTS_KERNEL: ${{ needs.set-driver-version-matrix.outputs[env.VAR_NAME_LTS_KERNEL] }}